tools/lib/src/integration_test_command.dart

```diff
@@ -70,25 +89,184 @@ class IntegrationTestCommand extends PluginCommand {
   @override
   String get name => 'integration-test';
 
-  File get _pythonTool => packagesDir.parent
-      .childDirectory('tools')
-      .childDirectory('tools')
-      .childFile('run_command.py');
-
+  // TODO(HakkyuKim): Consider validating command-line arguments in a parent
+  // object that subclasses [PackageLoopingCommand].
   @override
-  Future<void> run() async {
-    if (!_pythonTool.existsSync()) {
-      print('Error: Cannot find ${_pythonTool.path}.');
-      throw ToolExit(1);
+  Future<PackageResult> runForPackage(RepositoryPackage package) async {
+    // Copied to local variable to enable non-nullable type promotion.
+    final ArgResults? args = argResults;
+    if (args != null) {
```

Comment:
The example in [the doc I linked](https://dart.dev/guides/language/effective-dart/usage#consider-copying-a-nullable-field-to-a-local-variable-to-enable-type-promotion) is just a workaround. You can copy the value to a local `ArgResults` if you're sure that the value is non-null.

```suggestion
    final ArgResults args = argResults!;
```